drv/medusa-seq-server/src/power_control.rs

```diff
@@ -0,0 +1,159 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A crate for managing the power supplies on the Medusa board
+
+use crate::*;
+use drv_stm32xx_sys_api as sys_api;
+use sys_api::{OutputType, Port, Pull, Speed, Sys};
+
+task_slot!(SYS, sys);
+
+pub struct PowerRail {
+    /// The output GPIO for the power rail's enable pin
+    enable: sys_api::PinSet,
+    /// The input GPIO for the power rail's power good pin
+    power_good: sys_api::PinSet,
+    /// A RailName variant for ringbuf activity
+    name: RailName,
+}
+
+impl PowerRail {
+    pub fn new(
+        enable: sys_api::PinSet,
+        power_good: sys_api::PinSet,
+        name: RailName,
+    ) -> Self {
+        let sys = Sys::from(SYS.get_task_id());
+
+        sys.gpio_configure_output(
+            enable,
+            OutputType::PushPull,
+            Speed::Low,
+            Pull::None,
+        );
+
+        sys.gpio_configure_input(power_good, Pull::None);
+
+        Self {
+            enable,
+            power_good,
+            name,
+        }
+    }
+
+    /// Sets the enable pin for the power rail to HIGH if `enabled` is true or
+    /// LOW if it is false. If `enabled` is true, this function will also
+    /// monitor the power good signal for the supply.
```

Comment:
It does not! An earlier iteration of the function did, but I pulled out the monitoring to be its own thing because I did not want to wait around for monitoring on each individual power supply enable as that would be problematic when there's a need for multiple supplies to come up within a certain time of one another. I've removed the stale portion of the comment.